### PR TITLE
Drain node action

### PR DESCRIPTION
**Pull request by jakewins**
Three commits:

- Refactor actions into smaller files (because Drain is big)
- Drain action
- Init feature to allow actions a one-time init; drain uses this to clean up crashes

### CODE REVIEW

File: chaoskube/action/node_drain.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	if victim.Labels == nil {
+		victim.Labels = make(map[string]string, 0)
+	}
+	victim.Labels[LabelMarmosetCordoned] = "true"
+	victim.Spec.Unschedulable = true
+	return client.CoreV1().Nodes().Update(victim)
+}
+
+func uncordonNode(client kubernetes.Interface, victim *v1.Node) error {
+	delete(victim.Labels, LabelMarmosetCordoned)
```
Comment (jgrobbel) [inline chaoskube/action/node_drain.go]: Should the label not be cleared last?

**Review by jgrobbel — APPROVED** [review]
LGTM bar one question on order the removal of the label.

File: chaoskube/action/node_drain.go (hunk 1 above)
Comment (jakewins) [reply chaoskube/action/node_drain.go]: It's done atomically with the uncordon, so the node is uncordoned and unlabelled at the same time. That way we remove one possible in-between state we could end up in.

Comment (jgrobbel) [reply chaoskube/action/node_drain.go]: Ah its all  being applied through the update, for some reason thought `delete` was a method operating on the node. LGTM then 👍

**Pull request merged**